recipes/m4/all/test_package/conanfile.py

```diff
@@ -2,36 +2,40 @@
 from conans.errors import ConanException
 from io import StringIO
 import os
-
-
-M4_CONTENTS = """\
-m4_define(NAME1, `Harry, Jr.')
-m4_define(NAME2, `Sally')
-m4_define(MET, `$1 met $2')
-MET(`NAME1', `NAME2')
-"""
+import textwrap
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "arch", "compiler", "build_type"
 
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
     @property
     def _m4_input_path(self):
         return os.path.join(self.build_folder, "input.m4")
 
     def build_requirements(self):
-        if tools.os_info.is_windows:
-            self.build_requires("msys2/20200517")
+        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
+            self.build_requires("msys2/cci.latest")
+        if hasattr(self, "settings_build"):
+            self.build_requires(str(self.requires["m4"]))
 
     def build(self):
-        tools.save(self._m4_input_path, M4_CONTENTS)
+        tools.save(self._m4_input_path, textwrap.dedent("""\
+            m4_define(NAME1, `Harry, Jr.')
+            m4_define(NAME2, `Sally')
+            m4_define(MET, `$1 met $2')
+            MET(`NAME1', `NAME2')
+        """))
 
     def test(self):
         m4_bin = tools.get_env("M4")
```

Comment:
Guess we can only test `M4` when not cross building, and just use `m4` in other cases.